src/contracts/QVAULT.h

```
@@ -14,639 +44,2127 @@ struct QVAULT : public ContractBase
+
+                INVOKE_OTHER_CONTRACT_PROCEDURE(QX, AddToBidOrder, locals.addToBidOrder_input, locals.addToBidOrder_output, locals.addToBidOrder_input.price * locals.addToBidOrder_input.numberOfShares);
+
+                state.fundForBurn -= locals.addToBidOrder_input.price * locals.addToBidOrder_input.numberOfShares;
+                state.burntQcapAmPerEpoch.set(qpi.epoch(), state.burntQcapAmPerEpoch.get(qpi.epoch()) + (uint32)locals.addToBidOrder_input.numberOfShares);
+                state.totalQcapBurntAmount += (uint32)locals.addToBidOrder_input.numberOfShares;
+
+                locals.purchasedQcap += (uint32)locals.addToBidOrder_output.addedNumberOfShares;
+            }
+            else 
+            {
+                break;
+            }
+        }
 
-	}
+        locals.transferShareManagementRights_input.asset.assetName = QVAULT_QCAP_ASSETNAME;
+        locals.transferShareManagementRights_input.asset.issuer = state.QCAP_ISSUER;
+        locals.transferShareManagementRights_input.newManagingContractIndex = SELF_INDEX;
+        locals.transferShareManagementRights_input.numberOfShares = locals.purchasedQcap;
+
+        INVOKE_OTHER_CONTRACT_PROCEDURE(QX, TransferShareManagementRights, locals.transferShareManagementRights_input, locals.transferShareManagementRights_output, 0);
+
+        qpi.transferShareOwnershipAndPossession(QVAULT_QCAP_ASSETNAME, state.QCAP_ISSUER, SELF, SELF, locals.purchasedQcap, NULL_ID);
+    }
 
     struct END_EPOCH_locals 
     {
+        QX::TransferShareManagementRights_input transferShareManagementRights_input;
+		QX::TransferShareManagementRights_output transferShareManagementRights_output;
+        GPInfo updatedGProposal;
+        QCPInfo updatedQCProposal;
+        IPOPInfo updatedIPOProposal;
+        QEarnPInfo updatedQEarnProposal;
+        FundPInfo updatedFundProposal;
+        MKTPInfo updatedMKTProposal;
+        AlloPInfo updatedAlloProposal;
+        MSPInfo updatedMSProposal;
         ::Entity entity;
```

According to the current guidelines, the global-scope "::" is forbidden. Either use "QPI::" or remove it (and ignore the error that is incorrectly shown by the IDE).

---

fixed it.